expressEngine/lib/parser/base.js

```diff
@@ -11,6 +11,19 @@ class Parser {
   }
 
   response = (res, data, statusCode = 200) => res.status(statusCode).json(data);
+
+  validateRequired = (data = {}, required = []) => {
+    const retVal = [];
+    required.forEach((field) => {
+      if (!(data[field] || data[field] === false)) {
+        const fieldKey = field.charAt(0).toUpperCase() + field.slice(1);
+        retVal.push(`${fieldKey} is required.`);
+      }
+    });
+    if (retVal && retVal.length) {
+      throw new Error(`${retVal[0]}`);
```

Comment:
This will only throw the first missing value. Then there is no use of collecting all the missing values in the above loop.
This method can be improved 
`let missingValues = required.filter(field => !data[field]);
let error = "";
missingVvalues.forEach(item => error.push(getFirstCap(item)));
error += "are required"`

Reply:
Ok, I am updating error response method to send all errors in key/value pair

Reply:
Ok, I am updating error response method to send all errors in key/value pair